### ISSUE TITLE
Unclear usage for HopCroft Karp

### ISSUE THREAD

**Issue by GaurangTandon** [AnimeshSinha1309/algorithms-notebook#4]
It would be great if there could be few comments here and there on how to use the HopcroftKarp code for someone who knows what the algorithm is and how it works. Specifically, please explain: `size_u, size_v, nil, pair_u, pair_v, dist`.

**Comment by AnimeshSinha1309** [comment]
Yeah, well it was the size of both the sets u and v, and the guy who is pair with each element of u in v and v in u. `nil` and `dist` I am not sure of, I will check and add those as comments. If you can come up with better names please do.
